app/Actions/InstallUpdate/Pipes/AllowMigrationCheck.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Actions\InstallUpdate\Pipes;
+
+use App\Models\Configs;
+use App\Models\Logs;
+use Illuminate\Support\Facades\Config;
+
+class AllowMigrationCheck extends AbstractUpdateInstallerPipe
+{
+	public const ERROR_MSG =
+		/* @lang text */
+		'Update not applied: `APP_ENV` in `.env` is `production` and `force_migration_in_production` is set to `0`.';
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function handle(array &$output, \Closure $next): array
+	{
+		if (Config::get('app.env') !== 'production') {
+			return $next($output);
+		}
+
+		// @codeCoverageIgnoreStart
+		// we cannot code cov this part. APP_ENV is dev in testing mode.
```

Comment:
```suggestion
		// we cannot code cov this part. APP_ENV is `testing` in testing mode.
```
That's the correct one